awscli/customizations/cloudformation/modules.py

```diff
@@ -0,0 +1,525 @@
+            self.overrides = module_config[OVERRIDES]
+
+        # Resources defined in the module
+        self.resources = {}
+
+        # Parameters defined in the module
+        self.params = {}
+
+        # TODO: What about Conditions, Mappings, Outputs?
+        # Is there a use case for importing those into the parent?
+
+    def __str__(self):
+        "Print out a string with module details for logs"
+        return (
+            f"module name: {self.name}, "
+            + f"source: {self.source}, props: {self.props}"
+        )
+
+    def process(self):
+        """
+        Read the module source process it.
+
+        :return: The modified parent template dictionary
+        """
+
+        content = read_source(self.source)
+
+        module_dict = yamlhelper.yaml_parse(content)
+        if RESOURCES not in module_dict:
+            # The module may only have sub modules in the Modules section
+            self.resources = {}
+        else:
+            self.resources = module_dict[RESOURCES]
+
+        if PARAMETERS in module_dict:
+            self.params = module_dict[PARAMETERS]
+
+        # Recurse on nested modules
+        base_path = os.path.dirname(self.source)
+        section = ""
+        try:
+            section = MODULES
+            module_dict = process_module_section(module_dict, base_path)
+            section = RESOURCES
+            module_dict = process_resources_section(module_dict, base_path)
+        except Exception as e:
+            traceback.print_exc()
+            msg = f"Failed to process {section} section: {e}"
+            raise exceptions.InvalidModuleError(msg=msg)
+
+        self.validate_overrides()
+
+        for logical_id, resource in self.resources.items():
+            self.process_resource(logical_id, resource)
+
+        return self.template
+
+    def validate_overrides(self):
+        "Make sure resources referenced by overrides actually exist"
+        for logical_id in self.overrides:
+            if logical_id not in self.resources:
+                msg = f"Override {logical_id} not found in {self.source}"
+                raise exceptions.InvalidModuleError(msg=msg)
+
+    def process_resource(self, logical_id, resource):
+        "Process a single resource"
+
+        # For each property (and property-like attribute),
+        # replace the value if it appears in parent overrides.
+        attrs = [
+            PROPERTIES,
+            CREATIONPOLICY,
+            METADATA,
+            UPDATEPOLICY,
+            DELETIONPOLICY,
+            CONDITION,
+            UPDATEREPLACEPOLICY,
+            DEPENDSON,
+        ]
+        for a in attrs:
+            self.process_overrides(logical_id, resource, a)
+
+        # Resolve refs, subs, and getatts
+        #    (Process module Parameters and parent Properties)
+        container = {}
+        container[RESOURCES] = self.resources
+        self.resolve(logical_id, resource, container, RESOURCES)
```

Comment:
The container is only there because the recursive function needs to look back two levels to set d[n], so I needed a top level dictionary for the first iteration.